DllImportGenerator/DllImportGenerator/Marshalling/MarshallingGenerator.cs

```diff
@@ -122,6 +124,47 @@ internal class MarshallingGenerators
             TypePositionInfo info,
             StubCodeContext context)
         {
+            IMarshallingGenerator generator = CreateCore(info, context);
+            ValidateByValueMarshalKind(context, info, generator);
```

Comment:
Thoughts on making this a gate? I am suggesting that we convert this to a pipeline style function call so it becomes `return ValidateByValueMarshalKind(context, info, generator);`. Which means `ValidateByValueMarshalKind` would return the `IMarshallingGenerator` instance.

Reply:
May I assume it is just easier to to do all these checks here as opposed to pushing these checks deeper into the create pipeline?

Reply:
I can change this into a gate. I'm fine with that and it will be a little cleaner.

It's much easier doing the checks here than in every single marshaller, especially since we share marshaller instances for the simple marshallers.